tests/mypy_test.py

```diff
@@ -480,6 +482,11 @@ def test_third_party_stubs(code: int, args: TestConfig, tempdir: Path) -> TestRe
         if spec_matches_path(gitignore_spec, distribution_path):
             continue
 
+        dist_metadata = read_metadata(distribution)
+        if dist_metadata.requires_python and not dist_metadata.requires_python.contains(PYTHON_VERSION):
```

Comment:
Now I think we should look at both `args.version` and `parse_metadata.PYTHON_VERSION`. We want to skip the test either when we are running against an unsupported version, or because mypy and the package and its dependencies need to be installed on an supported version.

Reply:
Hmm, good point. Your logic makes sense to me. Would be nice, if possible, to tailor the message printed to the terminal to explain exactly why we're not checking these stubs in this specific situation, though

Reply:
BTW in your example, you used `requires_python = ">=3.8"` and `python tests/mypy_test.py -p3.8 stubs/requests`. I expect it not to skip the test in this case. Maybe you meant `python tests/mypy_test.py -p3.7 stubs/requests`?

Reply:
> Maybe you meant `python tests/mypy_test.py -p3.7 stubs/requests`

yes :)